pkg/providers/vmware/templates/template-finder.go

```diff
@@ -40,8 +39,7 @@ func (f *TemplateFinder) FindTemplate(vm *mo.VirtualMachine) (*templatev1.Templa
 	// We update metadata from the source vm so we default to medium flavor
 	namespace := TemplateNamespace
 	flavor := defaultFlavor
-	workload := defaultWorkload
-	tmpls, err := f.templateProvider.Find(&namespace, &os, &workload, &flavor)
+	tmpls, err := f.templateProvider.Find(&namespace, &os, nil, &flavor)
```

Comment:
With `nil` workload the provider will return multiple matching templates, from which only the newest would be picked. I think that resigning from controlling the workload type will lead to less predictable behaviour and application of suboptimal templates (with CPU flags tuned to a completely different workload) to imported VMs. 
I can imagine that user adds a new "desktop"-like template for a certain OS to the configuration and later all VMware imports for that OS use that template, even if most of them require `server`-type optimizations.

`server` is generally a quite safe default and I know that it has similar (but probably less severe) issues as described above (i.e. performance of Windows desktop VMs might suffer).

Maybe introduction of an optional `workload` property to the VirtualMachineImport would help here? It would give the user control over the workload type, if needed.

@pkliczewski, @machacekondra, @masayag, @fdupont-redhat - what do you think? Maybe it's actually not a real issue.

Reply:
I agree with @jakub-dzon. It is important to use proper workload type since optimizations part of the template are based on the workload type. I like the idea with workload property or having default workloads for specific OS types.

Reply:
The problem with setting `server` as the default is that it fails to find appropriate images for e.g. windows 10, which may only have server templates. I can search for a `server` template first, and fall back to searching for a `desktop` one if that fails. Is that more acceptable?

Reply:
@pkliczewski @jakub-dzon  as a stopgap until a workload property can be added or default workloads can be determined for each OS, would it be acceptable to default to looking for a `server` template, and then fall back to looking for a `desktop` template if nothing is returned?

Reply:
@mansam I think we could try with this approach. @jakub-dzon what do you think?

Reply:
> Maybe introduction of an optional `workload` property to the VirtualMachineImport would help here? It would give the user control over the workload type, if needed.

I'd prefer exposing template identifier (name + namespace) instead of specific properties for identifying the template.
Else, we'd end up with exposing the templates attributes used as field selectors.

Without having a template specified by the user, @mansam's approach to use `server` and to fallback to `desktop` sounds good to me.

Reply:
I'm okay with that approach. Moti's suggestion to pass template identifier instead of workload makes more sense than just the workload; it can provide easy workaround for any potential template resolution issues.

Reply:
As far as I remember we were explicitly asked not to use the name. @omeryahud Do you have any suggestions how to handle this?

Reply:
> As far as I remember we were explicitly asked not to use the name. @omeryahud Do you have any suggestions how to handle this?

Yes, using template names is generally not a good idea, because the names of the latest templates may change as new template bundles are released, and if you allow the user to choose the template used, they might not pick the latest template.

If you decide to accept template properties from the user, it should be the os/workload/flavor labels.

Just FYI, we are currently working on some things that were mentioned here:

- Default template for each OS - https://issues.redhat.com/browse/CNV-6153
- Fixing template names - https://issues.redhat.com/browse/CNV-6624